flask_potion/schema.py

```diff
@@ -256,10 +256,13 @@ def convert(self, instance, update=False, pre_resolved_properties=None, patchabl
     def parse_request(self, request):
         if request.method in ('POST', 'PATCH', 'PUT', 'DELETE'):
             if self.fields and request.mimetype != 'application/json':
-                raise RequestMustBeJSON()
+                # Allow when all fields are optional
+                if not all(((i.default is not None) or i.nullable
```

Comment:
Could you instead put the whole check into a new cached property, `is_optional` so that it does not have to be recomputed with every request?

Reply:
Good idea. Do you think it would make more sense to call it `all_optional` or ``all_are_optional``? To me, `is_optional` feels like it should be for a single field.

Reply:
Ok, let's go with `all_fields_optional`